pkg/network/common/dns.go

```diff
@@ -218,6 +291,9 @@ func (d *DNS) queryServer(nameserver, domain string) ([]net.IP, int, error) {
 }
 
 func (d *DNS) getIPsAndMinTTL(domain string) ([]net.IP, time.Duration, error) {
```

Comment:
`d.getIPsAndMinTTL` uses `d.nameservers` and `d.queryServer`. But it is not protected by a lock. It is probably OK since `d.nameservers` is not something that gets mucked around at all once the DNS object is created. And `d.queryServer` is not using any other shared resource.

Reply:
> d.getIPsAndMinTTL uses d.nameservers and d.queryServer. But it is not protected by a lock

If we locked the whole function we wouldn't be able to query the DNS in parallel, making the whole patch useless.  If there were a bit unsafe to call concurrently we should lock only that bit;  the rest must be unlocked so that we can call it concurrently.

> It is probably OK since d.nameservers is not something that gets mucked around at all once the DNS object is created. And d.queryServer is not using any other shared resource.

I agree, I don't think we need to lock it at all.